### PR TITLE
Implement metadata cascade policy support

### PR DESCRIPTION
**Pull request by mattwiller**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

File: docs/metadata.md [hunk 1]
```diff
@@ -635,3 +635,143 @@ client.folders.deleteMetadata('67890', client.metadata.scopes.GLOBAL, client.met
...
+[get-cascade-policies]: http://opensource.box.com/box-node-sdk/jsdoc/Metadata.html#getCascadePolicies
+
+Force Apply Cascade Policy
+--------------------------
+
+To force apply a metadata cascade policy and apply metadata values to all existing items in the affected
+folder, call [`metadata.forceApplyCascadePolicy(policyID, resolutionMethod, callback)`][force-apply-cascade]
+with the ID of the cascade policy to force apply and the conflict resolution method for dealing with items that
+already have a metadata value that conflicts with the folder.  Speicifying a resolution value of `'none'` will
+preserve the existing values on items, anmd specifying `'overwrite'` will overwrite values on the items in the
```
Comment (carycheng) [inline docs/metadata.md]: type on "anmnd"

File: docs/metadata.md [hunk 2]
```diff
@@ -635,3 +635,143 @@ client.folders.deleteMetadata('67890', client.metadata.scopes.GLOBAL, client.met
...
+
+[get-cascade-policies]: http://opensource.box.com/box-node-sdk/jsdoc/Metadata.html#getCascadePolicies
+
+Force Apply Cascade Policy
+--------------------------
+
+To force apply a metadata cascade policy and apply metadata values to all existing items in the affected
+folder, call [`metadata.forceApplyCascadePolicy(policyID, resolutionMethod, callback)`][force-apply-cascade]
+with the ID of the cascade policy to force apply and the conflict resolution method for dealing with items that
+already have a metadata value that conflicts with the folder.  Speicifying a resolution value of `'none'` will
```
Comment (carycheng) [inline docs/metadata.md]: Typo at "Speicifying"

File: docs/metadata.md [hunk 3]
```diff
@@ -635,3 +635,143 @@ client.folders.deleteMetadata('67890', client.metadata.scopes.GLOBAL, client.met
...
+			owner_enterprise: {
+				type: 'enterprise',
+				id: '11111'
+			},
+			parent: {
+				type: 'folder',
+				id: '22222'
+			},
+			scope: 'enterprise_11111',
+			templateKey: 'testTemplate'
```
Comment (carycheng) [inline docs/metadata.md]: Super minor but should this match the param 'myTemplate' passed in?

**Pull request merged**